server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -254,6 +254,20 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         Property.NodeScope
     );
 
+    // settings to configure maximum slice created per search request using OS custom slice computation mechanism. Default lucene
+    // mechanism will not be used if this setting is set with value > 0
+    public static final String CONCURRENT_SEGMENT_SEARCH_TARGET_MAX_SLICE_COUNT_KEY = "search.concurrent.max_slice";
```

Comment:
Nitpick, maybe include "count" in the settings key: `"search.concurrent.max_slice_count"`

Reply:
Will change this in 2.x branch as well as part of separate PR.